File: frontend/src/data/ImageController.ts
import { uploadImage } from "../http/Image";

export const UploadImageController = async (file: File) => {
    try {
        const image = await uploadImage(file);
        if (typeof image === "object" && image !== null && 'error' in image) {
            return {
                data: null,
                error: image.error,
                status: 500
            }
        }
        return {
            data: image,
            error: "",
            status: 200
        }
    } catch (error) {
        return {
            error: "Se produjo un error al subir la imagen: " + error,
        }
    }
};

File: backend/src/main/java/com/example/backend/controllers/ImageController.java
package com.example.backend.controllers;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;
import org.springframework.web.multipart.MultipartFile;
import com.example.backend.services.CloudinaryService;


@RestController
@RequestMapping("/images")
public class ImageController {

    @Autowired
    private CloudinaryService cloudinaryService;

    @PostMapping("/upload")
    public ResponseEntity<?> uploadImage(@RequestBody MultipartFile file) {
        try {
            String url = cloudinaryService.uploadImage(file);
            return ResponseEntity.ok().body("Imagen subida correctamente: " + url);
        } catch (Exception e) {
            return ResponseEntity.status(500).body("Error al subir imagen: " + e.getMessage());
        }
    }
}

